compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2907,7 +2932,7 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
      *           None         if the match fails and we should consider the following cases
      *                        because scrutinee and pattern do not overlap
```

Comment:
Very nice change switching to `MatchResult` - easier that re-reading this doc every time (at least for me 😅).  Could you update or remove this comment, please?